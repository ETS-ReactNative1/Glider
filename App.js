/**
 * @format
 * @flow strict-local
 */

import 'react-native-gesture-handler';
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { SafeAreaProvider, SafeAreaView } from 'react-native-safe-area-context';
import {
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

import Header from './app/shared/Header';
import HomeScreen from './app/screens/HomeScreen';

const Stack = createStackNavigator();

const App: () => React$Node = () => {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <StatusBar barStyle="light-content" />
        <SafeAreaView style={{ flex: 1, backgroundColor: '#FFA640'}} edges={['top']}>
          <Header/>
          <Stack.Navigator headerMode="none">
            <Stack.Screen name="Home" component={HomeScreen}/>
          </Stack.Navigator>
        </SafeAreaView>
      </NavigationContainer>
    </SafeAreaProvider>
  );
};

const styles = StyleSheet.create({
  sampleText: {
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: "center"
  }
});

export default App;
