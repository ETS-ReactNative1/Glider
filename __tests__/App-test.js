/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Main from '../app/screens/HomeScreen'

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

describe('React app snapshot test ', () => {

  it('App renders correctly', () => {
    const fullAppTest = TestRenderer.create(<App />).toJSON();
    expect(fullAppTest).toMatchSnapshot();
  });

  it('Homescreen renders correctly', () => {
    const fullAppTest = TestRenderer.create(<Main />).toJSON();
    expect(fullAppTest).toMatchSnapshot();
  });

})
