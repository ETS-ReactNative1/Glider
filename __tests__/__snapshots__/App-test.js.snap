// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app snapshot test  App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`React app snapshot test  Homescreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFA640",
        "height": "14%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "SourceSansPro-Black",
          "fontSize": 38,
        }
      }
    >
      Good Morning
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": "2%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Loading location...
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#462716",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 20,
          "paddingLeft": "5%",
          "paddingTop": "4%",
        }
      }
    >
      Where would you like to go? 
    </Text>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingTop": 15,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#462716",
                "fontFamily": "SourceSansPro-Bold",
                "fontSize": 16,
              }
            }
          >
            From
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#E98918",
                "borderBottomWidth": 1,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Enter starting point"
              placeholderTextColor="#D6A882"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "auto",
                  "color": "#462716",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 18,
                  "marginLeft": 10,
                  "marginTop": 5,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingTop": 15,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#462716",
                "fontFamily": "SourceSansPro-Bold",
                "fontSize": 16,
              }
            }
          >
            To
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#E98918",
                "borderBottomWidth": 1,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Enter destination"
              placeholderTextColor="#D6A882"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "auto",
                  "color": "#462716",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 18,
                  "marginLeft": 10,
                  "marginTop": 5,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "paddingRight": "15%",
            "paddingTop": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E98918",
                "borderColor": "#E98918",
                "borderRadius": 20,
                "borderWidth": 2,
                "height": 35,
                "justifyContent": "center",
                "marginTop": 20,
                "width": 100,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "SourceSansPro-Regular",
                  "fontSize": 18,
                }
              }
            >
              Search
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 6,
          "marginTop": "2%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
